src/librustc_mir/borrow_check/nll/constraint_generation.rs

```diff
@@ -164,7 +165,9 @@ impl<'cg, 'cx, 'gcx, 'tcx> Visitor<'tcx> for ConstraintGeneration<'cg, 'cx, 'gcx
                 self.location_table.mid_index(location),
             ));
 
-            for successor_block in terminator.successors() {
+            let successor_blocks = terminator.successors();
+            all_facts.cfg_edge.reserve(successor_blocks.size_hint().0);
```

Comment:
Could presumably `count()` too since I think it should be cheap (`successors` returns `Chain<IntoIter, slice::Iter>`)

Reply:
Neat, I didn't know it was this cheap for `slice::Iter`.

Reply:
I'm fairly certain it is (count() = len())